selfdrive/car/honda/hondacan.py

```diff
@@ -28,12 +28,22 @@ def make_can_msg(addr, dat, idx, alt):
   return [addr, 0, dat, alt]
 
 
-def create_brake_command(packer, apply_brake, pcm_override, pcm_cancel_cmd, chime, fcw, idx):
+def create_brake_command(packer, apply_brake, pcm_override, pcm_cancel_cmd, chime, fcw, car_fingerprint, idx):
   """Creates a CAN message for the Honda DBC BRAKE_COMMAND."""
+  commands = []
   pump_on = apply_brake > 0
   brakelights = apply_brake > 0
   brake_rq = apply_brake > 0
   pcm_fault_cmd = False
+  bus = 0
+
+  if car_fingerprint == CAR.CLARITY:
+    bus = 2
+    # This a bit of a hack but clarity brake msg flows into the last byte so
+    # rather than change the fix() function just set accordingly here.
+    apply_brake >>= 1
```

Comment:
What is the logic in this snippet? This would effectively halve apply_brake value
Also it seems message flows 2 bits into last byte. Why only a single bit is masked into idx (i.e. 0x8)

Reply:
wish i could tell you. i asked in the clarity channel to see if i get a response. still new to the OP thing, i have had mine for 3 weeks.

Reply:
It's OK. that question is intended at @energee or anyone maintaining a Clarity fork based on this logic (But I guess all other forks copied this logic from here)

Reply:
This has been in since the first Clarity ports @etorgersen do you remember who put this in?

Reply:
Looking at the commit history it was added by @energee

Reply:
From discussing it in the discord channel it was added because of a peculiarity with the Honda Clarity, without that bit right before a full stop the breaks would momentarily let up and them apply again at full force. I don't know if this is because of the regenerative braking. According to @etorgersen the fix was the modifying the apply brake message

Reply:
Hmm quite interesting.
In fact I have that very bad behavior (But somewhat attenuated and repeated) still even after this fix in my Honda Fit Hybrid JDM which uses same message structure.
I guess I would do more tests with stock system and see.
I thought that has something to do with tuning (PID saturation or something) but still can't figure out why it's happening or how this fix supposedly solves it. (I mean exact mechanism/logic behind)

Reply:
I was looking at a incidence where this behavior happening and it seems to relate to more how OP handles pump_on request than the requested brake value. It seems at least on Fit Hybrid JDM (Which is similar to Clarity with respect to brake message, pump need to stay on while braking.

![Annotation 2019-09-27 232942](https://user-images.githubusercontent.com/11025494/65791158-c5130e00-e17e-11e9-8f8e-6d0b9c8897a8.jpg)